docs/tutorials-samples/targeting-core-for-new-libraries.md

```diff
@@ -0,0 +1,298 @@
+                "System.Runtime":"4.0.0-rc1-*"
+            }
+        },
+        "dotnet55":{
+           "dependencies":{
+                "System.Runtime":"4.0.0-rc1-*"
+            }
+        },
+        ".NETPortable,Version=v4.0,Profile=Profile328":{
+            "frameworkAssemblies":{
+                "mscorlib":"",
+                "System":"",
+                "System.Core":""
+            }
+        }
+    }
+}
+```
+
+This will allow you to target .NET Core 5.1, .NET Core 5.5, .NET Framework 4.0, Windows 8, Windows Phone 8.1, WIndows Phone Silverlight 8.1, and Silverlight 5.0.
+
+Run `dnu restore` and `dnu build` from the command line, and your library can now be built!  You will now notice three new entries in your `/bin/Debug` folder:
+
+```
+/bin
+   /Debug
+      /dotnet51
+      /dotnet55
+      /portable-net40+sl50+netcore45+wpa81+wp8
+```
+
+Finally, run `dnu pack` to build a NuGet pakage, and your `/bin/Debug` folder should look like this:
+
+```
+/bin
+   /Debug
+      /dotnet51
+      /dotnet55
+      /portable-net40+sl50+netcore45+wpa81+wp8
+      Lib.1.0.0.nupkg
+      Lib.1.0.0.symbols.nupkg
+```
+
+And now you can publish a NuGet package!
+
+**NOTE:** This assumes your code will compile across *both* .NET Core and .NET Framework.  Read the section on cross-compiling with `#if`s on how to compile the same file differently for each target if you are using features which are unavailable in some of your targets.
+
+## How do I cross-compile for .NET Core and .NET Framework?
+
+If you need to support .NET Framework and .NET Core, you may find yourself needing to use different APIs to accomplish the same task.  You may also wish to support different language constructs (such as `async` and `await`) and support modern APIs within the same file.  A great way to do this is to cross-compile with `#if` guards.
+
+For example, in .NET Core you can use `HttpClient` in the `System.Net.Http` API to perform network operations over Http.  However, the .NET 4.0 Framework does not include `HttpClient`!  Instead, you may need to use `WebClient` from the `System.Net` assembly.
+
+First, the `project.json` file should look something like this:
+
+```
+{
+    "frameworks":{
+        "net40":{
+            "frameworkAssemblies": {
+                "System.Net":"",
+                "System.Text.RegularExpressions":""
+            }
+            
+        },
+        "dotnet55":{
+            "dependencies": {
+                "System.Runtime":"4.0.0-rc1-*",
+                "System.Net.Http": "4.0.1-beta-23409",
+                "System.Text.RegularExpressions": "4.0.11-beta-23409",
+                "System.Threading.Tasks": "4.0.11-beta-23409"
+            }
+        }
+    }
+}
+
+```
+
+Note that framework assemblies being used are explicitly referenced in the `net40` target, and NuGet references are also explictly listed in the `dotnet55` target.
+
+Next, your `using`s in your source file can be adjusted like this:
+
+```csharp
+#if NET40
```

Comment:
Do we need to explain where the symbol is defined, and give examples of other useful ones?